management/src/main/java/com/upresent/management/service/GeoFenceServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package com.upresent.management.service;
+
+import java.util.Calendar;
+import java.util.List;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+import com.google.gson.Gson;
+import com.upresent.management.entity.GeoFenceData;
+import com.upresent.management.entity.UserDetail;
+import com.upresent.management.exception.ExceptionResponseCode;
+import com.upresent.management.exception.ManagementException;
+import com.upresent.management.producer.KafkaMessageProducer;
+import com.upresent.management.producer.RestMessageProducer;
+import com.upresent.management.repository.GeoFenceRepository;
+import com.upresent.management.repository.UserRepository;
+import com.upresent.management.requestdto.GeoFenceReq;
+import com.upresent.management.utils.CommonUtility;
+import com.upresent.management.utils.Constant;
+import com.upresent.management.utils.UserType;
+
+@Service
+public class GeoFenceServiceImpl implements GeoFenceService {
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Autowired
+	private KafkaMessageProducer kafkaMessageProducer;
+
+	@Autowired
+	private RestMessageProducer restMessageProducer;
+
+	@Autowired
+	private GeoFenceRepository geoFenceRepository;
+
+	@Autowired
+	private Environment env;
+
+	Gson gson = new Gson();
+
+	@Override
+	public String addGeoFence(GeoFenceReq geoFenceReq) throws ManagementException {
+		String username = geoFenceReq.getUsername();
+		if (CommonUtility.isValidString(username)) {
+			List<UserDetail> users = userRepository.findByUsername(username);
```

Comment:
We should make a call to user service and remove the userRepository implementation in management service.